src/builder.ts

```diff
@@ -565,25 +594,98 @@ export class SchemaBuilder {
       return true
     }
 
-    const message = `${typeName}.${fieldName} is referencing a type "${outputType}" that is not defined in your GraphQL schema.`
+    const message = `Your GraphQL \`${typeName}\` object definition is projecting a field \`${fieldName}\` with \`${outputType}\` as output type, but \`${outputType}\` is not defined in your GraphQL Schema`
 
-    if (stage === 'build') {
-      if (isDevMode()) {
-        if (this.options.onUnknownFieldType) {
-          this.options.onUnknownFieldType({
-            unknownFieldType: outputType,
-            typeName,
-            fieldName,
-            error: new Error(message),
-          })
-        } else {
-          console.log(`Warning: ${message}`)
-        }
-      } else {
-        throw new Error(message)
-      }
-    }
+    registerHook(
+      this.options.onUnknownFieldType,
+      [
+        {
+          unknownFieldType: outputType,
+          typeName,
+          fieldName,
+          error: new Error(message),
+        },
+      ],
+      message,
+      stage,
+    )
 
     return false
   }
+
+  assertArgNameExists(
```

Comment:
With minor change we can extract this into a pure function? The more methods we accumulate the larger the class-state-mutating surface area becomes. 🤔